src/Interpreters/Session.cpp

```diff
@@ -439,11 +439,12 @@ ContextMutablePtr Session::makeQueryContextImpl(const ClientInfo * client_info_t
     ContextMutablePtr query_context = Context::createCopy(from_session_context ? session_context : global_context);
     query_context->makeQueryContext();
 
+    auto query_context_user = query_context->getUser();
     LOG_DEBUG(log, "{} Creating query context from {} context, user_id: {}, parent context user: {}",
               toString(auth_id),
               from_session_context ? "session" : "global",
               user_id ? toString(*user_id) : "<EMPTY>",
-              query_context->getUser() ? query_context->getUser()->getName() : "<NOT SET>");
+              query_context_user ? query_context_user->getName() : "<NOT SET>");
```

Comment:
Messages like `Creating query context from session context, user_id: <NOT SET>` in logs look creepy (because session cannot be created without a user AFAIK). We should avoid such messages. And especially we should avoid messages like `Authenticated with global context as user <EMPTY>`. See also #30370

Reply:
Well, first, this is a debug message (hence not really intended for the general public). Second, it just states how things are. But Ok, I removed the message if you don't like it.

Reply:
> Well, first, this is a debug message (hence not really intended for the general public)

Default log level in clickhouse is `trace` (yep, that's weird, but it works this way historically), it includes debug messages as well. We have `test` log level for messages that not intended for the general public.

> Second, it just states how things are. But Ok, I removed the message if you don't like it.

The problem is that things should not be like this and that's what I actually don't like. We should either save `UserPtr` somewhere to be able to print meaningful messages or throw exception if user is required but it's nullptr. Maybe it's ok to throw exception later, but at least comment is required in this case. Now this code looks like we may allow some unknown user to start a session without authentication.

And what about `Authenticated with global context as user <EMPTY>`? It's much more creepy, because it makes you think that someone has hacked your server. (and the same question about other messages added in #30370, I wouldn't merge it if I was a reviewer).

Reply:
Btw, I've just found these WTF-messages in server log on a production cluster:
```
2022.05.10 11:43:16.938357 [ 414 ] {} <Debug> TCP-Session: a0e10978-38c5-496c-bfe1-b5b270cb169a Destroying unnamed session of user <EMPTY>
2022.05.10 11:43:17.790144 [ 414 ] {} <Information> TCPHandler: Client has not sent any data.
2022.05.10 11:43:17.790156 [ 414 ] {} <Debug> TCPHandler: Done processing connection.
2022.05.10 11:43:17.790174 [ 414 ] {} <Debug> TCP-Session: a9eb1685-eae7-45ce-948e-3f72fa7faab8 Destroying unnamed session of user <EMPTY>
2022.05.10 11:43:17.993485 [ 414 ] {} <Information> TCPHandler: Client has not sent any data.
2022.05.10 11:43:17.993496 [ 414 ] {} <Debug> TCPHandler: Done processing connection.
2022.05.10 11:43:17.993512 [ 414 ] {} <Debug> TCP-Session: 642b742a-3037-4293-a6b5-4fe1cf14541a Destroying unnamed session of user <EMPTY>
2022.05.10 11:43:18.476784 [ 414 ] {} <Information> TCPHandler: Client has not sent any data.
2022.05.10 11:43:18.476795 [ 414 ] {} <Debug> TCPHandler: Done processing connection.
```